src/test/java/io/vertx/rabbitmq/RabbitMQServiceTest.java

```diff
@@ -31,43 +34,40 @@
   @Override
   public void setUp() throws Exception {
     super.setUp();
-
-    if ("true".equalsIgnoreCase(System.getProperty("rabbitmq.local"))) {
-      client = RabbitMQClient.create(vertx, config());
-      CountDownLatch latch = new CountDownLatch(1);
-      client.start(onSuccess(v -> {
-        latch.countDown();
-      }));
-      awaitLatch(latch);
-      channel = new ConnectionFactory().newConnection().createChannel();
-    } else {
-      // Use CloudAMQP
-      RabbitMQOptions config = config().setUri(CLOUD_AMQP_URI);
-      client = RabbitMQClient.create(vertx, config);
-      CountDownLatch latch = new CountDownLatch(1);
-      client.start(onSuccess(v -> {
-        latch.countDown();
-      }));
-      awaitLatch(latch);
-      ConnectionFactory factory = new ConnectionFactory();
-      factory.setUri(CLOUD_AMQP_URI);
-      channel = factory.newConnection().createChannel();
-    }
   }
 
-
   @Override
   protected void tearDown() throws Exception {
-    channel.close();
+    if(channel != null) channel.close();
     super.tearDown();
   }
 
   public RabbitMQOptions config() {
     return new RabbitMQOptions();
   }
 
+  @Rule
+  public SniffyRule sniffyRule = new SniffyRule();
+
+  @Test
+  @DisableSockets
+  public void testStartWithReconnectFuture() throws Exception {
+    Future<Void> future = Future.future();
+    CountDownLatch latch = new CountDownLatch(1);
+    RabbitMQOptions options = "true".equalsIgnoreCase(System.getProperty("rabbitmq.local")) ? config().setUri(CLOUD_AMQP_URI) : config();
```

Comment:
Shouldnt it be : 

"true".equalsIgnoreCase(System.getProperty("rabbitmq.local")) ? config() : config().setUri(CLOUD_AMQP_URI)